# robot_localization EKF configuration
# See: https://docs.ros.org/en/rolling/p/robot_localization/
ekf_filter_node:
  ros__parameters:
    frequency: 30.0
    sensor_timeout: 0.1
    two_d_mode: true  # Operating in 2D mode as the robot moves in a planar environment, despite IMU supporting 9DOF
    publish_tf: true
    odom_frame: odom
    base_link_frame: base_footprint
    world_frame: map
        # Odometry (replace with your actual odom topic if different)
    odom0: /r2/odom
    odom0_config: [true,  true,  false,  # x, y, z position
                   true, true, false, # x, y, z velocity
                   false, false, true,  # roll, pitch, yaw orientation
                   false, false, true, # x, y, z linear accel
                   false, false, false] # x, y, z angular velocity
    odom0_differential: false
    odom0_relative: false
    odom0_queue_size: 10

    # IMU
    imu0: /r2/imu/data_raw
    imu0_config: [false, false, false,  # x, y, z position
                  false, false, false,  # x, y, z velocity
                  false, false, false,  # roll, pitch, yaw orientation
                  true, true, true,  # x, y, z linear accel
                  true, true, false]  # x, y, z angular velocity
    imu0_queue_size: 10
    imu0_differential: false
    imu0_relative: true
    imu0_remove_gravitational_acceleration: true
    # Output
    print_diagnostics: true
